blt/src/Blt/Plugin/Commands/HsCommands.php

```diff
@@ -8,13 +8,73 @@
 use Drupal\Core\Serialization\Yaml;
 use GuzzleHttp\Client;
 use Robo\Exception\TaskException;
+use Sws\BltSws\Blt\Plugin\Commands\SwsCommandTrait;
 
 /**
  * Various BLT commands for H&S stack.
  */
 class HsCommands extends BltTasks {
 
-  use HsCommandTrait;
+  use SwsCommandTrait;
+
+  /**
+   * After code deployed, update all sites on the stack.
+   *
+   * @command humsci:post-code-deploy
+   *
+   * @aliases humsci:post-code-update
+   */
+  public function postCodeDeployUpdate() {
+    $sites = $this->getConfigValue('multisites');
+    $parallel_executions = 10;
+    $site_chunks = array_chunk($sites, ceil(count($sites) / $parallel_executions));
+    $commands = [];
+    foreach ($site_chunks as $sites) {
+      $commands[] = $this->blt()
+        ->arg('humsci:update-sites')
+        ->arg(implode(',', $sites))
+        ->getCommand();
+    }
+    file_put_contents(sys_get_temp_dir() . '/update-report.txt', '');
+    $this->taskExec(implode(" &\n", $commands) . PHP_EOL . 'wait')->run();
+    $report = array_filter(explode("\n", file_get_contents(sys_get_temp_dir() . '/update-report.txt')));
+
+    $success = [];
+    $failed = [];
+    foreach ($report as $line) {
+      [$site, $status] = explode(':', $line);
+      if ((int) $status) {
+        $success[] = $site;
+      }
+      else {
+        $failed[] = $site;
+      }
+    }
+    $this->yell(sprintf('Updated %s sites successfully.', count($success)), 100);
+    if ($failed) {
+      $this->yell(sprintf("Update failed for the following sites:\n%s", implode("\n", $failed)), 100, 'red');
+    }
+    unlink(sys_get_temp_dir() . '/update-report.txt');
+  }
+
+  /**
+   * @command humsci:update-sites
+   *
+   * @var string $sites
+   *   List of sites to update.
+   */
+  public function updateSites($sites = NULL) {
+    $sites = $sites ? explode(',', $sites) : $this->getConfigValue('multisites');
+    foreach ($sites as $site_name) {
+      $this->switchSiteContext($site_name);
+      $result = $this->taskDrush()
+        ->drush('updb')
+        ->drush('config:import')
+        ->option('partial')
+        ->run();
+      file_put_contents(sys_get_temp_dir() . '/update-report.txt', $site_name . ($result->wasSuccessful() ? ':1' : ':0') . PHP_EOL, FILE_APPEND);
```

Comment:
This assumes the `update-report.txt` file exists, but it's created in the post deploy command and this command can be run separately.

Reply:
Would it hard fail if it didn't exist?

Reply:
Nevermind we're good:

https://www.php.net/manual/en/function.file-put-contents.php

> If filename does not exist, the file is created